src/together/resources/finetune.py

```diff
@@ -125,11 +138,76 @@ def createFinetuneRequest(
         wandb_project_name=wandb_project_name,
         wandb_name=wandb_name,
         train_on_inputs=train_on_inputs,
+        from_checkpoint=from_checkpoint,
     )
 
     return finetune_request
 
 
+def _process_checkpoints_from_events(
+    events: List[FinetuneEvent], id: str
+) -> List[FinetuneCheckpoint]:
+    """
+    Helper function to process events and create checkpoint list.
+
+    Args:
+        events (List[FinetuneEvent]): List of fine-tune events to process
+        id (str): Fine-tune job ID
+
+    Returns:
+        List[FinetuneCheckpoint]: List of available checkpoints
+    """
+    checkpoints: List[FinetuneCheckpoint] = []
+
+    for event in events:
+        event_type = event.type
+
+        if event_type == FinetuneEventType.CHECKPOINT_SAVE:
+            step = get_event_step(event)
+            checkpoint_name = f"{id}:{step}" if step is not None else id
+
+            checkpoints.append(
+                FinetuneCheckpoint(
+                    type=(
+                        f"Intermediate (step {step})"
+                        if step is not None
+                        else "Intermediate"
+                    ),
+                    timestamp=event.created_at,
+                    name=checkpoint_name,
+                )
+            )
+        elif event_type == FinetuneEventType.JOB_COMPLETE:
+            if hasattr(event, "model_path"):
+                checkpoints.append(
+                    FinetuneCheckpoint(
+                        type=(
+                            "Final Merged"
+                            if hasattr(event, "adapter_path")
+                            else "Final"
+                        ),
+                        timestamp=event.created_at,
+                        name=id,
+                    )
+                )
+
+            if hasattr(event, "adapter_path"):
+                checkpoints.append(
+                    FinetuneCheckpoint(
+                        type=(
+                            "Final Adapter" if hasattr(event, "model_path") else "Final"
+                        ),
+                        timestamp=event.created_at,
+                        name=id,
+                    )
+                )
+
+    # Sort by timestamp (newest first)
+    checkpoints.sort(key=lambda x: x.timestamp, reverse=True)
```

Comment:
Looks like we're comparing timestamps as strings here, which can produce [unexpected results](https://stackoverflow.com/questions/68339815/why-does-sorting-strings-with-numbers-have-a-different-result-than-just-sorting). Maybe `timestamp` should be a datetime object or at least an integer? This way, you also don't need to parse them inside `format_timestamp` (which also contradicts the point of the function somewhat)

Reply:
Looks like this is not strictly necessary, because our REST API returns timestamps in [RFC3339](https://www.rfc-editor.org/rfc/rfc3339.html) that was designed to have chronological order same as the lexicographic order. So you can treat this as a nit, even though our logic here becomes somewhat implementation-specific :)

Reply:
I would like to keep it as it's now :)